thewalrus/tests/test_entanglement.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Entanglement tests"""
+# pylint: disable=no-self-use, assignment-from-no-return
+import pytest
+
+import numpy as np
+
+from thewalrus.quantum import entanglement_entropy, log_negativity
+from thewalrus.random import random_interferometer
+from thewalrus.symplectic import (
+    interferometer,
+    passive_transformation,
+    two_mode_squeezing,
+    xxpp_to_xpxp,
+)
+
+
+def random_cov(num_modes=200, pure=False, nonclassical=True):
+    r"""Creates a random covariance matrix for testing.
```

Comment:
wonder if it is adding this a full fledged function in the random module

Reply:
You mean in this PR, a parallel one, or a future one?

Reply:
I guess all options are valid. But given that this is in some sense an orthogonal functionality a future PR would be more appropriate.